internal/exec/util/passwd.go

```diff
@@ -171,16 +178,24 @@ func (u Util) AuthorizeSSHKeys(c types.PasswdUser) error {
 		if !strings.HasSuffix(ks, "\n") {
 			ks = ks + "\n"
 		}
-
+		var path string
 		if distro.WriteAuthorizedKeysFragment() {
-			err = writeAuthKeysFile(usr, filepath.Join(usr.HomeDir, ".ssh", "authorized_keys.d", "ignition"), []byte(ks))
+			path = filepath.Join(usr.HomeDir, ".ssh", "authorized_keys.d", "ignition")
+			err = writeAuthKeysFile(usr, path, []byte(ks))
 		} else {
-			err = writeAuthKeysFile(usr, filepath.Join(usr.HomeDir, ".ssh", "authorized_keys"), []byte(ks))
+			path = filepath.Join(usr.HomeDir, ".ssh", "authorized_keys")
+			err = writeAuthKeysFile(usr, path, []byte(ks))
 		}
-
 		if err != nil {
 			return fmt.Errorf("failed to set SSH key: %v", err)
 		}
+		if err == nil {
```

Comment:
should we just make this an else case for the previous if statement?

Reply:
I can drop `if` statement instead and put that logic after `if err !=nil` condition statement. If `err==nil` then it will eventually execute the later part for the code.